File: lei_lds2122_grupo2-main/Frontend/src/app/viewModels/requests/LoginVM.ts
import { toJSONString } from "../interfaces/toJSONString"

/**
 * View Model for a login request
 */
export class LoginVM implements toJSONString {

    /**
     * Construct's an instance of {@link LoginVM}
     * 
     * @param email user's email
     * @param password user's password
     */
    constructor(
        public email: string,
        public password: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    public toJSONString(): string {
        return JSON.stringify(this)
    }
}

File: Backend/Data/ViewModels/Request/LoginVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View model class for login request
    /// </summary>
    public class LoginVM
    {
        public string Email{ get; set; }
        public string Password{ get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="email">User's email</param>
        /// <param name="password">User's password</param>
        public LoginVM(string email, string password)
        {
            this.Email = email;
            this.Password = password;
        }
    }
}